examples/vpc-scenario-2/main.tf

```diff
@@ -60,24 +70,25 @@ module "ubuntu-xenial-ami" {
   release = "14.04"
 }
 
-resource "aws_key_pair" "main" {
-  key_name   = "${var.name}"
-  public_key = "${file(var.ssh_pubkey)}"
+module "vpc-sg" {
+  source      = "../../modules/security-group-base"
+  description = "Test project security group"
+  name        = "${var.name}-vpc-sg"
+  vpc_id      = "${module.vpc.vpc_id}"
 }
 
 # shared security group for SSH
 module "public-ssh-sg" {
   source              = "../../modules/ssh-sg"
-  name                = "${var.name}"
-  vpc_id              = "${module.vpc.vpc_id}"
-  allowed_cidr_blocks = "0.0.0.0/0"
+
+  security_group_id = "${module.vpc-sg.id}"
 }
 
 # shared security group, open egress (outbound from nodes)
 module "open-egress-sg" {
```

Comment:
I don't think this is happening elsewhere, but we should probably consider updating our scheme to include `rule` in the modules that are adding rules. We would then rename these modules from `open-egress-sg` to `open-egress-rule` or `open-egress-sg-rule`. I'm not sure if it makes sense to do that here and now, or wait and do that across the board all in one fell swoop.

The context here: we used to provide a set of modules that created their own security group, which meant attaching multiple SG to a single instance. There's a limit on the number of SG you can attach (6, I think), so that eventually didn't work. Also, TF added support for "creating a SG, then attached N rules to the SG", and we decided to refactor our modules around the concept.. so in the initial 0.7.x release, we refactored the SG modules to provide a "base SG" and then other modules to "attach rules" to that base SG.

Reply:
@ketzacoatl I think this relates to what you mentioned here. Since there is no longer a `vpc-sg` module and no longer a `vpc-sg.id` value to set the `security_group_id` to for the open-egress-sg and public-ssh-sg modules, in my next commit I set `security_group_id` to `vpc-elb-sg.id`.